instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/server/NettyServerErrorHandler.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.netty.v3_8.server;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.ErrorCauseExtractor;
+
+public final class NettyServerErrorHandler {
+
+  // copied from BaseTracer#onException()
+  public static void onError(Context context, Throwable error) {
```

Comment:
Is this needed? Shouldn't `Instrumenter` take care of that?

Reply:
Alternatively could add a context key that can hold this exception and extract it for calling end. Idk whether it would be nicer.

Reply:
That'd be a pretty good idea. I'll do that for all netty versions in a separate PR (they all handle exceptions this way currently)